### ISSUE TITLE
Add support for preprocessed columns

### ISSUE THREAD

**Issue by hecmas** [0xPolygonMiden/air-script#135]
To make AirScript constraint scope capture [randomized AIR with preprocessing (RAP)](https://hackmd.io/@aztec-network/plonk-arithmetiization-air) (and not limiting it to AIR with injected verifier randomness), it would be necessary to add support for preprocessed columns. This would make AirScript a DSL ready to be used in other projects, since RAPs are the most generic way we have at the moment to capture computations of a computational model. For example, RAPs can easily describe Plonk or R1CS constraint systems, making it able to capture arithmetic circuits.

Based on the idea of how periodic columns are handled in AirScript and knowing that preprocessed columns are publicly known to both the prover and the verifier before the STARK generation, it makes sense to invent a syntax for preprocessed columns that is explicit. 

I think that one of the two following approaches would work (I use python-like syntax to denote the preprocessed columns shape):
1. Add a  new declaration for preprocessed columns in the `trace_columns` section, e.g.:
```
trace_columns:
    main: [a,b]
    aux: [c,d]
    pre:
        L1: [1] + [0]*(T-1)
```

2. Create a section for preprocessed columns, e.g.:
```
preprocessed_columns:
        L1: [1] + [0]*(T-1)
```
I the previous examples, I have used `T`for denoting the number of rows in the execution trace.

**Comment by hecmas** [comment]
I completely agree that keeping the number of rows out of AirScript syntax is a good idea. 

I am less sure how you could keep maintaining that in the XOR example: the `xor.csv` would need to somehow have a number of rows equal to the expected number of rows in the execution trace (and if you decide to change it, then you would also need to change it i the `xor.csv` file). Anyways, I agree that AirScript need some mechanism to handle preprocessed columns that are not succinct.

**Comment by jan-ferdinand** [comment]
The number of rows could be different. For example, consider a scenario with only two tables. As long as the xor table is
1. the shorter one, and
2. contains and “end-of-table” marker that is set in every row after the, well, end of the table,

the lookup can work exactly the same way. Of course, looking up a row that is beyond the end of the table must fail.
